lib/route/nh.c

```diff
@@ -240,6 +240,24 @@ int rtnl_nh_get_fdb(struct rtnl_nh *nexthop)
 	return nexthop->ce_mask & NH_ATTR_FLAG_FDB;
 }
 
+int rtnl_nh_set_family(struct rtnl_nh *nexthop, uint8_t family)
+{
+	if (!nexthop)
+		return -NLE_INVAL;
+
+	nexthop->nh_family = family;
+
+	return 0;
```

Comment:
seems `nh_compare()` ignores `nh_family`. That should be fixed... (not necessarily by you).